### ISSUE TITLE
Env vars depending on others not always set

### ISSUE THREAD

**Issue by MolotovCherry** [sagiegurari/cargo-make#934]
### Describe The Bug
An env var in the `[env]` block that depends on another var doesn't always properly set the value

### To Reproduce
I'm in a workspace, I don't know if that has anything to do with it or not. I'm also on Windows 11 x64, using cargo-make 0.36.13

Run the provided configuration many times `cargo make -p prod bar`.
Sometimes it will display `2`, other times it will display `${VAR2}`.

```toml
[env]
VAR1="${VAR2}"

[env.prod]
VAR2=2

[env.devel]
VAR2=3

[tasks.bar]
script = ['''
    echo %VAR1%
    ''']

[config]
default_to_workspace = false
```

Note: My actual use-case is a script env var in `[env]`.

Also note: I just started using this today, my usage may not necessarily be correct as I'm still learning. If I made a mistake, let me know. In either case, I still can't get it to work properly for some reason

**Comment by MolotovCherry** [comment]
I'm able to work around it for now by doing something like
```toml
[tasks.set-path]
env = { "VAR" = { script = ['''
    echo some result
'''] } }

[tasks.bar]
script = ['''
    echo %VAR%
    ''']
```

**Comment by sagiegurari** [comment]
i'll investigate...

**Comment by MolotovCherry** [comment]
I'll try to make a small repro sample later today when I'm a little more free 😄

**Comment by MolotovCherry** [comment]
> i'll investigate... but if you can provide a test case like a small repo or something that i can use to reproduce, it would help.

I made an example repo which reproduces it for me. Turns out there wasn't much needed to reproduce at all. I used my original example

https://github.com/MolotovCherry/cargo-make-934

**Comment by sagiegurari** [comment]
and sometimes it works and sometimes it doesn't?

**Comment by MolotovCherry** [comment]
> and sometimes it works and sometimes it doesn't?

Yeah. If I run it enough times, it'll display `${VAR2}` or `2`. Usually about 50/50, but sometimes there's "luck" and it'll have a series of successes

**Comment by sagiegurari** [comment]
@MolotovCherry i'm able to reproduce this.

**Comment by sagiegurari** [comment]
@MolotovCherry it is now fixed in the 0.36.14 dev branch. however, i don't think you are going to be that happy.
i fixed that strict order is kept (issue was use of hashset instead of indexset to check circular depedencies).
this does mean however, that var1 is using var2 before var2 is defined which ends up undefined each time.
to resolve, you need to put var1 inside the profile block as well

```toml
[env]
VAR1 = "any"

[env.prod]
VAR2 = 2
VAR1 = "${VAR2}"

[env.devel]
VAR2 = 3
VAR1 = "${VAR2}"
```

**Comment by MolotovCherry** [comment]
> @MolotovCherry it is now fixed in the 0.36.14 dev branch. however, i don't think you are going to be that happy. i fixed that strict order is kept (issue was use of hashset instead of indexset to check circular depedencies). this does mean however, that var1 is using var2 before var2 is defined which ends up undefined each time. to resolve, you need to put var1 inside the profile block as well
> 
> ```toml
> [env]
> VAR1 = "any"
> 
> [env.prod]
> VAR2 = 2
> VAR1 = "${VAR2}"
> 
> [env.devel]
> VAR2 = 3
> VAR1 = "${VAR2}"
> ```

Thank you for the quick fix and and explanation! 😄

**Comment by sagiegurari** [comment]
new cargo-make is now released.
